juicefs-1.3.yaml

```diff
@@ -0,0 +1,139 @@
+  - name: ${{package.name}}-compat
+    description: Compat package for ${{package.name}}
+    dependencies:
+      runtime:
+        - lsof # Check which process is using a file, mount point, or socket.
+        - procps # Check running JuiceFS processes
+        - strace # Analyze what a JuiceFS mount process is doing at the syscall level
+        - iproute2 # Check interface and routing info if JuiceFS
+        - net-tools # Check open ports or connections
+        - openssh-server
+        - openssh-client
+      provides:
+        - juicefs-compat=${{package.full-version}}
+    pipeline:
+      - runs: |
+          mkdir -p "${{targets.contextdir}}"/usr/local/bin
+          mkdir -p "${{targets.contextdir}}"/bin
+          ln -sf /usr/bin/juicefs "${{targets.contextdir}}"/usr/local/bin/juicefs
+          ln -sf /usr/bin/juicefs "${{targets.contextdir}}"/bin/mount.juicefs
+    test:
+      pipeline:
+        - runs: |
+            test "$(readlink /usr/local/bin/juicefs)" = "/usr/bin/juicefs"
+            test "$(readlink /bin/mount.juicefs)" = "/usr/bin/juicefs"
+
+update:
+  enabled: true
+  ignore-regex-patterns:
+    - '-rc'
+    - '-beta'
+    - '-dev'
+  github:
+    identifier: juicedata/juicefs
+    strip-prefix: v
+    tag-filter: v1.3.
+
+test:
+  environment:
+    environment:
+      MINIO_ACCESS_KEY: minioadmin
+      MINIO_SECRET_KEY: minioadmin
+      ACCESS_KEY: minioadmin
+      SECRET_KEY: minioadmin
+      MINIO_ROOT_USER: minioadmin
+      MINIO_ROOT_PASSWORD: minioadmin
+      WEBDAV_USER: minioadmin
+      WEBDAV_PASSWORD: minioadmin
+    contents:
+      packages:
+        - redis
+        - minio
+        - mc
+        - wait-for-it
+        - curl
+  pipeline:
+    - name: "Version test"
+      runs: |
+        juicefs --version 2>&1 | grep -q ${{package.version}}
+    - name: "Start redis for metadata"
+      runs: |
+        redis-server --daemonize yes > /tmp/redis.logs 2>&1 &
+        sleep 2
+    - name: "Start MinIO"
+      runs: |
+        mkdir -p /data/minio
+        minio server /data/minio --address :9000 --console-address :9001 > /tmp/minio.log 2>&1 &
+        wait-for-it localhost:9000 -t 30
+    - name: "Create test bucket using mc"
+      runs: |
+        mc alias set local http://localhost:9000 minioadmin minioadmin > /tmp/mc.log 2>&1
+        mc mb local/juicefs-bucket 2>&1 | grep -q "Bucket created successfully"
+    - name: "Format juicefs volume and dump logs"
+      runs: |
+        juicefs format --bucket http://localhost:9000/juicefs-bucket --access-key $ACCESS_KEY --secret-key $SECRET_KEY redis://localhost testvolume 2>&1 | grep -q "Volume is formatted"
+        juicefs dump redis://localhost /tmp/dump.json
+        grep -q "Sustained" /tmp/dump.json
+        grep -q "DelFiles" /tmp/dump.json
+        grep -q "FSTree" /tmp/dump.json
+    - name: "Test S3 Gateway setup"
+      runs: |
+        # Setup S3 Gateway
+        juicefs gateway redis://localhost localhost:9002 > gateway.logs 2>&1 &
```

Comment:
Any reasons why we don't use `test/daemon-check-output` pipeline here?

Reply:
We had a combined test step for all service commands earlier (s3 gateway and webdav) and recently split it in different test blocks recently.
I've updated tests with using `daemon-check-output`